cpp/ql/test/query-tests/Likely Bugs/Format/NonConstantFormat/test.cpp

```diff
@@ -41,14 +41,18 @@ const char *const_wash(char *str) {
 }
 
 int main(int argc, char **argv) {
+  const char *message = messages[2];
   printf(choose_message(argc - 1), argc - 1); // OK
+  printf(messages[1]); // OK
+  printf(message); // OK
   printf(make_message(argc - 1)); // NOT OK
+  printf("Hello, World\n"); // OK
   printf(_("Hello, World\n")); // OK
   {
     char hello[] = "hello, World\n";
     hello[0] = 'H';
     printf(hello); // NOT OK
-    printf(_(hello)); // NOT OK
+    printf(_(hello)); // OK
```

Comment:
I agree with the way you think here: everything that comes out of `gettext` is safe (for the purpose of this query).

Reply:
The query actually goes a bit further and assumes that `_(...)` is always OK, regardless of where the `_` macro points at.  Do you think this is overly lenient?

Reply:
We could look into adjusting it, but I'd prefer keeping this PR focused on the data flow change.

Reply:
Ok